diff --git a/device/archived/linux-samsung-a5y17lte/APKBUILD b/device/archived/linux-samsung-a5y17lte/APKBUILD
index b92b44b78..5521cb018 100644
--- a/device/archived/linux-samsung-a5y17lte/APKBUILD
+++ b/device/archived/linux-samsung-a5y17lte/APKBUILD
@@ -34,6 +34,7 @@ source="
 	debug_muic.patch
 	s2muu004-reduce-chatter.patch
 	fix-check-lxdialog.patch
+	fix-tcp-quickack-linkage.patch
 "
 builddir="$srcdir/$_repository-$_commit"
 _qcaclddir="$srcdir/quic-la-platform-vendor-qcom-opensource-wlan-qcacld-2.0-$_qca_commit"
@@ -94,7 +95,7 @@ package() {
 sha512sums="
 f3aaeb94386d6847c2407bc89b2a23c083324ed1c7cf095cbd40a72335efe642d4bbc755848e508b9caa09d0c75adb2064134461ed5e1d7d1a76b8c7007c8a4b  linux-samsung-a5y17lte-725aa5ab01e5a1272f34f01764f09009edf32777.tar.gz
 1dfcf7d1965d1a4fa86ac05d4408f1ffbedaf864689e0228d09d3d964fca08edf33dcf77bb0ab4dfcf1f8a37d6ee300c9e480457710745f9f5b7df9587e265e8  qcacld-48f9ab5d5c161549be098997fa0e3602c1547213.tar.gz
-932081de999b2dd4307a6e1f5c7bc958cbe9b7c2bf2ce1750d1cde58daa54adee52ca2352469bbc136e726beada86cd35ddd73f5fe39b382ceaa9d4c209f00d9  config-samsung-a5y17lte.aarch64
+ac238d3965ce5409efd00c2756cdc61333463b2ea9735e29caa9a5edd24a0df94dc16da78e506ffeca7a317e4781c0f98715367da55bcbd3982260121742e4dc  config-samsung-a5y17lte.aarch64
 9b13cccc9323f343d0aa676de6c6a83f34815cf06656adfa7ba20b559d94fc46c04b1508f1db22d12a14d4cdd531994f548b77a4a4d7d07481ee88a56564c0c9  fix_firmware_path.patch
 75545ccd0082a119b9ad95fcef3523c939a5c946f8f3a5c3dde15cee1502f6c1011050a0ad5359426f018bb7ceea1c97a28c7665941bc89f365c8d11065ce0f6  linux3.7to4.5-ARM64-remove-Sun-Solaris-style-flag-on-section.patch
 b5f7aed0d4c78fcd9ea6f65e11606c4d33004b45eed39017baa8b0a78db801396aa04ba7cd3a4a669c23f878f5462ae1b5f753e0e26b65c9962fd1e98855e524  0007-usb_gadget-set-random-rndis-host-MAC-address-to-prev.patch
@@ -105,4 +106,5 @@ f763033e73f5a00ad42273c11058446ff365d09d5f204c828d111aa48bc816baca74c333cbd2fa61
 d40153e1d86d407301ff3e6824b8445bf259bffdd50179149f67012b42829e565ce9dbeeeb2eca34bb14ddabcc694dd28793e6faa4f44a88f962e33490766ff7  debug_muic.patch
 399e5ab5f4ff662b9f892c351373bd1fb535461c7f230c531378436e9c9f61429a90447c0f5ce1b872a026b03c0104c9483e9cb2577ba044e5415da805deda39  s2muu004-reduce-chatter.patch
 182be3c596b9cc267ac108d7cf03fc8c328ccc6b36770800e4dcedea8d1bb65e3f5eacf590c2948f58b1418cc60a1670ba77dde8c259e428d158c31b6e1dbaf5  fix-check-lxdialog.patch
+cb0e4b6e398428e5a2f952e0f08565d8a5601cdf01caa7ffd0d68c7621a92773fb0d3dbd07a8a45f5099a421f094d9981936160a323dbb14eb83b9e7700486c4  fix-tcp-quickack-linkage.patch
 "
diff --git a/device/archived/linux-samsung-a5y17lte/config-samsung-a5y17lte.aarch64 b/device/archived/linux-samsung-a5y17lte/config-samsung-a5y17lte.aarch64
index 90463cf18..03b8a693e 100644
--- a/device/archived/linux-samsung-a5y17lte/config-samsung-a5y17lte.aarch64
+++ b/device/archived/linux-samsung-a5y17lte/config-samsung-a5y17lte.aarch64
@@ -631,11 +631,7 @@ CONFIG_IPV6_MULTIPLE_TABLES=y
 # CONFIG_IPV6_SUBTREES is not set
 # CONFIG_IPV6_MROUTE is not set
 # CONFIG_NETLABEL is not set
-CONFIG_MPTCP=y
-# CONFIG_MPTCP_PM_ADVANCED is not set
-CONFIG_DEFAULT_MPTCP_PM="default"
-# CONFIG_MPTCP_SCHED_ADVANCED is not set
-CONFIG_DEFAULT_MPTCP_SCHED="default"
+# CONFIG_MPTCP is not set
 # CONFIG_ANDROID_PARANOID_NETWORK is not set
 CONFIG_NET_ACTIVITY_STATS=y
 CONFIG_NETWORK_SECMARK=y
diff --git a/device/archived/linux-samsung-a5y17lte/fix-tcp-quickack-linkage.patch b/device/archived/linux-samsung-a5y17lte/fix-tcp-quickack-linkage.patch
new file mode 100644
index 000000000..12b911d6e
--- /dev/null
+++ b/device/archived/linux-samsung-a5y17lte/fix-tcp-quickack-linkage.patch
@@ -0,0 +1,12 @@
+--- a/net/ipv4/tcp_input.c
++++ b/net/ipv4/tcp_input.c
+@@ -191,9 +191,6 @@
+ 	icsk->icsk_ack.quick = quickacks;
+ }
+ 
+-#ifndef CONFIG_MPTCP
+-static 
+-#endif
+ void tcp_enter_quickack_mode(struct sock *sk, unsigned int max_quickacks)
+ {
+ 	struct inet_connection_sock *icsk = inet_csk(sk);
